packages/shell-api/src/replica-set.spec.ts

```diff
@@ -609,34 +608,43 @@ describe('ReplicaSet', () => {
     const delay = util.promisify(setTimeout);
     const ensureMaster = async(timeout): Promise<void> => {
       while (!(await rs.isMaster()).ismaster) {
-        if (timeout > 8000) {
-          return expect.fail(`Waited for ${host}:${port0} to become master, never happened`);
+        if (timeout > 32000) {
+          return expect.fail(`Waited for ${cfg.members[0].host} to become master, never happened`);
         }
         await delay(timeout);
         timeout *= 2; // try again but wait double
       }
       expect((await rs.conf()).members.length).to.equal(3);
     };
 
-    before(async() => {
-      serviceProvider = await CliServiceProvider.connect(connectionString);
+    before(async function() {
+      this.timeout(100_000);
+      const [ srv0, srv1, srv2, srv3 ] = await Promise.all(testServers);
+      cfg = {
+        _id: replId,
+        members: [
+          { _id: 0, host: `${srv0.host()}:${srv0.port()}`, priority: 1 },
+          { _id: 1, host: `${srv1.host()}:${srv1.port()}`, priority: 0 },
+          { _id: 2, host: `${srv2.host()}:${srv2.port()}`, priority: 0 }
+        ]
+      };
+      additionalServer = srv3;
+
+      serviceProvider = await CliServiceProvider.connect(srv0.connectionString());
       internalState = new ShellInternalState(serviceProvider);
       mongo = internalState.currentDb.getMongo();
       rs = new ReplicaSet(mongo);
 
       // check replset uninitialized
       try {
         await rs.status();
+        expect.fail();
```

Comment:
So mlaunch supports starting up uninitialized replsets? If it's not possible then we can potentially just skip the init test for now

Reply:
@aherlihy Yes, the tests here run & pass :) It doesn’t have a designated feature for starting uninitialized replsets, but you can pass pretty much any mongod server option through over it